storage/statedb/database.go

```diff
@@ -335,6 +335,18 @@ func NewDatabaseWithCache(diskDB database.DBManager, cacheSizeMB int) *Database
 	}
 }
 
+// NewDatabaseWithCache creates a new trie database to store ephemeral trie content
+// before its written out to disk or garbage collected. It also acts as a read cache
+// for nodes loaded from disk.
+func NewDatabaseWithCache(diskDB database.DBManager, cache *fastcache.Cache) *Database {
```

Comment:
`NewDatabaseWithCache` and `NewDatabaseWithNewCache` are actually doing a same thing with different configuration. What do you think about combining them into one?
```
// NewDatabaseWithNewCache creates a new trie database to store ephemeral trie content
// before its written out to disk or garbage collected. It also acts as a read cache
// for nodes loaded from disk.
// If both cache and cache size is given as parameters, the given cache will be used
// and cache size will be ignored. If neither of them is given, a Database without
// cache is created.
func NewDatabaseWithCache(diskDB database.DBManager, cache *fastcache.Cache, cacheSizeMB int) *Database {
	if cache == nil {
		if cacheSizeMB == AutoScaling {
			cacheSizeMB = getTrieNodeCacheSizeMB()
		}
		if cacheSizeMB > 0 {
			cacheSizeByte := cacheSizeMB * 1024 * 1024
			cache = fastcache.New(cacheSizeByte)

			logger.Info("Initialize trie node cache with fastcache", "MaxMB", cacheSizeMB)
		}
	}
	return &Database{
		diskDB:        diskDB,
		nodes:         map[common.Hash]*cachedNode{{}: {}},
		preimages:     make(map[common.Hash][]byte),
		trieNodeCache: cache,
	}
}
```

Reply:
As we talked in offline, I think simple 2 functions are better than 1 complex function. :)